bin/alpen-bridge/src/rpc_server.rs

```diff
@@ -181,25 +181,32 @@ impl StrataBridgeMonitoringApiServer for BridgeRpc {
         &self,
         deposit_request_outpoint: OutPoint,
     ) -> RpcResult<DepositRequestStatus> {
-        let result = self
-            .db
-            .get_deposit_request_by_txid(deposit_request_outpoint.txid)
-            .await
-            .map_err(|_| {
-                ErrorObjectOwned::owned::<_>(
-                    -666,
-                    "Database error. Config dumped",
-                    Some(self.db.config()),
-                )
-            })?;
-        match result {
-            Some(deposit) => Ok(deposit),
-            None => Err(ErrorObjectOwned::owned::<_>(
-                -32001,
-                "Deposit request outpoint not found",
-                Some(deposit_request_outpoint),
-            )),
+        // If it is Requested, then we can just return the InProgress status.
+        // Otherwise, it is complete.
+        if let Some(contract_sm) = self.current_state.get(&deposit_request_outpoint.txid) {
+            match contract_sm.get_state() {
```

Comment:
you can lift out the `return Ok(...)` to outside the match statement if you want and it may clean things up a bit.

Reply:
Also this is one of those situations where a `let...else` may be more appropriate.